public/components/custom_panels/helpers/utils.tsx

```diff
@@ -462,16 +542,43 @@ export const isPPLFilterValid = (
   return true;
 };
 
+export const processMetricsData = (schema: any, dataConfig: any) => {
+  if (isEmpty(schema)) return {};
+  if (
+    schema.length === 3 &&
+    schema.every((schemaField) => ['@labels', '@value', '@timestamp'].includes(schemaField.name))
+  ) {
+    return prepareMetricsData(schema, dataConfig);
+  }
+  return {};
+};
+
+export const prepareMetricsData = (schema: any, dataConfig: any) => {
+  const metricBreakdown: any[] = [];
```

Comment:
We should add types and interfaces for all these any types.

Reply:
These are also dynamic objects and there i no smaller type that covers.